exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/grpc/DefaultGrpcExporterBuilder.java

```diff
@@ -147,4 +164,25 @@ public DefaultGrpcExporterBuilder(
         stubFactory.apply(channel).withCompression(codec.getMessageEncoding());
     return new DefaultGrpcExporter<>(type, channel, stub, timeoutNanos, compressionEnabled);
   }
+
+  /**
+   * Reflectively access a {@link DefaultGrpcExporterBuilder} instance in field called "delegate" of
+   * the instance.
+   *
+   * @throws IllegalArgumentException if the instance does not contain a field called "delegate" of
+   *     type {@link DefaultGrpcExporterBuilder}
+   */
+  public static <T> DefaultGrpcExporterBuilder<?> getDelegateBuilder(Class<T> type, T instance) {
```

Comment:
should we scope `T` to at least have some bounds? Do we have a common interface we could enforce here to make it a bit clearer what this is for/about? This is a weird API to have be public at the moment.

Reply:
> should we scope T to at least have some bounds? Do we have a common interface we could enforce here to make it a bit clearer what this is for/about? 

Unfortunately not. 

> This is a weird API to have be public at the moment.

Yeah its not ideal, but at least its an internal package. There's tons of public stuff in the internal packages that is questionable if you view it as truly public. 

The alternative to this was to have this functionality package private in the autoconfigure module. I like this approach though because it allows someone to programmatically configure a retry policy that is different from the default one that is used when you enable it via env vars.

Reply:
Somehow I missed this was `internal`. Carry on!